### ISSUE TITLE
Speed-Up Plagiarism Detection

### ISSUE THREAD

**Issue by Garfield96** [ls1intum/Artemis#4766]
### Describe the bug

Currently, plagiarism detection is very slow, which sometimes causes problems (server not reachable, no result shown, etc.). The origin of the problem is twofold:
1. The download of all repositories happens sequential (related: https://github.com/ls1intum/Artemis/issues/1226)
2. [JPlag](https://github.com/jplag/JPlag) is run in normal mode instead of parallel mode.

### To Reproduce

Run plagiarism check on an exercise with many participants (600+).

### Expected behavior

Run all steps in parallel and thus speed up execution.

### What browsers are you seeing the problem on?

Firefox

### Additional context

JPlag currently only distinguishes between normal and parallel mode and does not allow the specification of a max. number of used processors. Thus, extending JPlag with an option to limit parallelism is desirable to avoid excessive resource consumption.

**Comment by krusche** [comment]
We will try to speedup the process, however please keep in mind that plagiarism checks have a lower priority than other processes on Artemis. Performing too many operations at the same time (e.g. while others have large exams) can lead to an overall unstable system performance.

I think we would need to synchronize plagiarism operations in a way that e.g. only one operation per course is running at the same time, but instructors could queue up operations and get notified when they are finished. In case of active exams in large courses, the operations could be performed after the exam has finished...

**Comment by krusche** [comment]
The general performance was improved the last couple of weeks.
